### PR TITLE
[ENHANCEMENT] log client ip to help identify which client is no org id

### PR DESCRIPTION
**Pull request by Komey**
**What this PR does**:
add log for client ip to help identify which client is no org id
**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by electron0zero — APPROVED** [review]

File: modules/distributor/receiver/middleware.go [hunk 1]
```diff
@@ -61,12 +61,14 @@ func (m *multiTenancyMiddleware) Wrap(next consumer.Traces) consumer.Traces {
 			info := client.FromContext(ctx)
 			orgIDs := info.Metadata.Get(user.OrgIDHeaderName)
 			if len(orgIDs) == 0 {
-				log.Logger.Log("msg", "failed to extract org id from both grpc and HTTP", "err", err)
+				log.Logger.Log("msg", "failed to extract org id from both grpc and HTTP",
+					"err", err, "client", info.Addr.String())
```
Comment (mapno) [inline modules/distributor/receiver/middleware.go:65]: This is causing a panic when `info.Addr == nil`. You need to add a check for that.

**Review by mapno — COMMENTED** [review]
LGTM, but the panic needs to be fixed.

File: modules/distributor/receiver/middleware.go (hunk 1 above)
Comment (Komey) [reply modules/distributor/receiver/middleware.go:65]: HI @mapno, 
Thanks for the reviews, for version [v0.74.0](https://github.com/grafana/tempo/blob/main/go.mod#LL73C13-L73C13) of the opentelemetry-collector SDK, `info.Addr` is not a pointer. we should pay attention to future updates.

https://github.com/open-telemetry/opentelemetry-collector/blob/v0.74.0/client/client.go#L104

Comment (mapno) [reply modules/distributor/receiver/middleware.go:65]: `net.Addr` is an interface, so it panics when undefined. See example https://go.dev/play/p/dx6a1BYB16J.

Comment (Komey) [reply modules/distributor/receiver/middleware.go]: @mapno Thanks for the correction, the panics has been fixed

**Review by mapno — APPROVED** [review]
LGTM, thank you for the contribution.

**Pull request merged**